src/features/all-drops/components/AllDrops.tsx

```diff
@@ -146,37 +134,30 @@ export default function AllDrops() {
 
     let nftHref = '';
     if (type === DROP_TYPE.NFT) {
-      const fcMethod = (fc as ProtocolReturnedFCData).methods[0]?.[0];
-      const { receiver_id } = fcMethod as ProtocolReturnedMethod;
-
-      const nftData = await asyncWithTimeout(
-        viewCall({
-          contractId: receiver_id,
-          methodName: FETCH_NFT_METHOD_NAME,
-          args: {
-            mint_id: parseInt(id),
-          },
-        }),
-      ).catch((_) => {
-        console.error(); // eslint-disable-line no-console
-      });
-
-      nftHref =
-        `${CLOUDFLARE_IPFS}/${nftData?.metadata?.media as string}` ??
-        'https://placekitten.com/200/300';
+      let nftMetadata = {
+        media: '',
+        title: '',
+        description: '',
+      };
+      try {
+        nftMetadata = await keypomInstance.getNftMetadata(drop);
+      } catch (e) {
+        console.error('failed to get nft metadata', e); // eslint-disable-line no-console
+      }
+      nftHref = nftMetadata?.media ?? 'https://placekitten.com/200/300';
```

Comment:
why we got a kitten image as fallback imag? 🤣 could we use image from `public/assets/not-found-image` instead?

also if `await keypomInstance.getNftMetadata(drop);` throws an error, `nftMetadata?.media` would be `''`, then the placeholder image won't show. Would suggest to use `||` instead of `??`